src/socket/tcp.rs

```diff
@@ -549,6 +557,16 @@ impl<'a> Socket<'a> {
         }
     }
 
+    /// Enable or disable TCP Timestamp.
+    pub fn set_tsval_generator(&mut self, generator: Option<fn() -> u32>) {
```

Comment:
Should this be a `fn` or `FnMut`?

Reply:
having `generator` an `FnMut` would allow closures, which would make it much more flexible, but will require every user of `socket` to have a generic.
Another option is to have it behind a `Box`, but that would be an indirection on a function that will be called a lot (every packet)
I don't have a strong opinion, if you think it should be one of those two options, that can be easily done

Reply:
having `generator` an `FnMut` would allow closures, which would make it much more flexible, but will require every user of socket to have a generic.
Another option is to have it behind a `Box`, but that would be an extra indirection on a function that will be called for every TCP packet.
I don't have a strong opinion, if you think I should change it to one of these options, that can be easily done

Reply:
Let's ship it as `fn()` and deal with it later if it becomes an issue.